packages/bundler-plugin-core/src/utils.ts

```diff
@@ -307,3 +308,27 @@ export function stripQueryAndHashFromPath(path: string): string {
   // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
   return path.split("?")[0]!.split("#")[0]!;
 }
+
+export function replaceBooleanFlagsInCode(
+  code: string,
+  values: Record<string, boolean | undefined>
+): { code: string; map: SourceMap } | null {
+  const ms = new MagicString(code);
+
+  Object.keys(values).forEach((key) => {
+    const value = values[key];
+
+    if (typeof value === "boolean") {
+      ms.replaceAll(key, value ? "true" : "false");
```

Comment:
```suggestion
      ms.replaceAll(key, JSON.stringify(value));
```

It's a bit more flexible but does the same thing.